packages/datetime/src/dateInput.tsx

```diff
@@ -447,16 +471,49 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
             // the page. tabbing forward should *not* close the popover, because
             // focus will be moving into the popover itself.
             this.setState({ isOpen: false });
+        } else if (e.which === Keys.ESCAPE) {
+            this.setState({ isOpen: false });
+            this.inputRef.blur();
         }
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
+    // blur DOM event listener (not React event)
+    private handlePopoverBlur = () => this.handleClosePopover();
+
+    private registerPopoverBlurHandler = () => {
+        if (this.contentRef != null) {
+            // Popover contents are well structured, but the selector will need
+            // to be updated if more focusable components are added in the future
+            const tabbableElements = this.contentRef.querySelectorAll("input, [tabindex]:not([tabindex='-1'])");
+            const numOfElements = tabbableElements.length;
+            if (numOfElements > 0) {
+                const lastPopoverElement = tabbableElements[numOfElements - 1] as HTMLElement;
+                if (this.lastPopoverElement !== lastPopoverElement) {
+                    this.unregisterPopoverBlurHandler();
+                    this.lastPopoverElement = lastPopoverElement;
+                    this.lastPopoverElement.addEventListener("blur", this.handlePopoverBlur);
```

Comment:
is this right? what if you TAB to the last element and then SHIFT+TAB to go backwards?

Reply:
Actually, if you SHIFT+TAB on the last element, it will close. Will fix.

- [ ] Handle SHIFT+TAB and make sure it focuses previous element.